### PR TITLE
feat: cloud source repo scanner

### LINKED ISSUE #19: Implement Cloud Source Repositories API crawler
[Cloud Source Repositories](https://cloud.google.com/source-repositories/docs/reference/rest/v1/projects.repos/list) allows to query list of repos per project. We need to implement a scanner capable of querying list of projects.

### PR DESCRIPTION
**Pull request by sudiptob2**
Fixes #19 

- [x] Implement cwaler
- [x] Add entry in the scanner.py
- [x] update example configuration
- [x] Add unittest 
- [x] Add dummy test file to compare against

The unit test will fail because the source repo hasn't been enabled in the `test-project for gcp-scanner` yet in google cloud.
Once the source repo is enabled, updating the `test/sourcerepo` file will fix the test.

### CODE REVIEW

**Comment by sudiptob2** [comment]
@mshudrak need a review on this PR

File: src/gcp_scanner/crawl.py [hunk 1]
```diff
@@ -1005,3 +1005,38 @@ def list_services(project_id: str, credentials: Credentials) -> List[Any]:
...
+    A list of cloud source repositories in the project.
+  """
+
+  logging.info("Retrieving cloud source repositories %s", project_id)
+  list_of_repos = list()
+  service = discovery.build("sourcerepo", "v1", credentials=credentials)
+
+  request = service.projects().repos().list(
+    name="projects/" + project_id,
+    pageSize=200
```
Comment (mshudrak) [inline src/gcp_scanner/crawl.py:1027]: Is 500 maximum? Let's try to fetch as much data as possible with one query.

File: test/sourcerepos [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (mshudrak) [inline test/sourcerepos:1]: I will update it manually once we merge this PR.

**Review by mshudrak — CHANGES_REQUESTED** [review]
That's great! There are a couple of small comments to address.

File: src/gcp_scanner/crawl.py (hunk 1 above)
Comment (sudiptob2) [reply src/gcp_scanner/crawl.py:1027]: Agreed, but I think this line handles the paginated response. So, if there are more repos, eventually that will be fetched. 
I took the example from `list_services` function which was previously implemented. My understanding could be wrong in this case, looking froward to your opinion.

File: src/gcp_scanner/test_unit.py [hunk 3]
```diff
@@ -400,3 +400,14 @@ def test_project_info(self):
         "project_info",
       )
     )
+
+  def test_sourcerepos(self):
+    """Test list of cloud source repositories in the project."""
+    # TODO: Enable cloud source repo in the test project
```
[the remark this answers is not in the document]
Comment (sudiptob2) [reply src/gcp_scanner/test_unit.py:406]: Thanks, will remove the unit test todods.

[the remark this answers is not in the document]
Comment (sudiptob2) [reply src/gcp_scanner/test_unit.py:406]: or should I omit the unit test entirely.

File: test/sourcerepos (hunk 2 above)
Comment (sudiptob2) [reply test/sourcerepos:1]: Understood, will remove this file.

File: src/gcp_scanner/test_unit.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mshudrak) [reply src/gcp_scanner/test_unit.py:406]: Nope, just TODO

File: test/sourcerepos (hunk 2 above)
Comment (mshudrak) [reply test/sourcerepos:1]: You can keep it, I will just update it.

Comment (sudiptob2) [reply test/sourcerepos:1]: understood, will remove the content of the file.

File: src/gcp_scanner/crawl.py (hunk 1 above)
Comment (mshudrak) [reply src/gcp_scanner/crawl.py:1027]: Yes, I know that eventually everything will be fetched, we just want to reduce overall amount ofl interaction (requests) with Cloud API.

Comment (sudiptob2) [reply src/gcp_scanner/crawl.py:1027]: great, understood clearly.

**Comment by mshudrak** [comment]
LGTM

**Pull request merged**